app/src/main/java/fr/free/nrw/commons/media/MediaClient.java

```diff
@@ -161,4 +155,29 @@ public MediaClient(MediaInterface mediaInterface) {
                 .map(Media::from)
                 .single(Media.EMPTY);
     }
+
+    public Single<String> getCaption(String filename)  {
+        return mediaDetailInterface.fetchCaptionByFilename(Locale.getDefault().getLanguage(), filename)
+                .map(mediaDetailResponse -> {
+                    if(mediaDetailResponse.getSuccess() == 1) {
+                        Map<String, CommonsWikibaseItem> entities = mediaDetailResponse.getEntities();
+                        if (entities != null) {
+                            Map.Entry<String, CommonsWikibaseItem> entry = entities.entrySet().iterator().next();
+                            CommonsWikibaseItem commonsWikibaseItem = entry.getValue();
+                            if (commonsWikibaseItem != null) {
+                                Map<String, Caption> labels = commonsWikibaseItem.getLabels();
+                                if (labels != null) {
+                                    Map.Entry<String, Caption> captionEntry = labels.entrySet().iterator().next();
+                                    Caption caption = captionEntry.getValue();
+                                    if (caption != null) {
+                                        return caption.getValue();
+                                    }
+                                }
+                            }
+                        }
+                    }
+                   return "NO CAPTION";
```

Comment:
If possible, remove the UI row instead.
Or at least, use detail_caption_empty.

Reply:
okay I'll do in the next commit